### PR TITLE
Add cross platform build for full .NET Framework

### PR DESCRIPTION
**Pull request by hazzik**

### CODE REVIEW

File: src/NHibernate.Test/NHibernate.Test.csproj [hunk 1]
```diff
@@ -58,7 +58,9 @@
   </ItemGroup>
   <ItemGroup Condition="'$(TargetFramework)'=='net461'">
     <Reference Include="System.Configuration" />
+    <Reference Include="System.Runtime" />
     <Reference Include="System.Transactions" />
+    <Reference Include="System.Threading.Tasks" />
```
Comment (hazzik) [inline src/NHibernate.Test/NHibernate.Test.csproj]: There is System.Runtime and System.Threading.Tasks that come from some ADO.NET provider (I think this is Npgsql) which do not match the ones from the `/Facades` folder, so reference them explicitly.

**Comment by ngbrown** [comment]
What is the incentive for this?

**Comment by hazzik** [comment]
There are a few reasons (not in any particular order):

- I use Mac a lot for development now days
- Unify build scripts:
  - Get rid of NAnt some day
  - Move some of the tests from TeamCity to AppVeyor
- Add tests on Mono

File: build-common/DotNetSdkMono.props [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <!-- When compiling .NET SDK 2.0 projects targeting .NET 4.x on Mono using 'dotnet build' you -->
+    <!-- have to teach MSBuild where the Mono copy of the reference asssemblies is -->
```
[the remark this answers is not in the document]
Comment (hazzik) [reply build-common/DotNetSdkMono.props]: Yes, the file is from somewhere else. But, I don't know the origin. It's originated somewhere [here](https://github.com/dotnet/sdk/issues/335) (most likely [this](https://github.com/dotnet/sdk/issues/335#issuecomment-368669050) comment). Most likely this is THE source of the file: https://github.com/Microsoft/visualfsharp/blob/cb9e692331c2f0d66b0f72d49a07533204011517/fcs/netfx.props

**Comment by hazzik** [comment]
> Is that the reason for this PR to remain a WIP?

No, the builds were failing because of different referenced versions of System.Runtime and System.Threading.Tasks. But since then I did not remove WIP label.

> this PR just test that net461 target do build under AppVeyor and Travis.

Yes, only for that for now. The full targets on AppVeyor and Travis will follow, but in separate PR's

**Pull request merged**